tests/unit/unit-walker.cpp

```diff
@@ -38,11 +45,62 @@ class UnitWalkerTests : public ::testing::Test,
     bvsort = s->make_sort(BV, 4);
     funsort = s->make_sort(FUNCTION, SortVec{ bvsort, bvsort });
     arrsort = s->make_sort(ARRAY, bvsort, bvsort);
+    boolsort = s->make_sort(BOOL);
   }
   SmtSolver s;
-  Sort bvsort, funsort, arrsort;
+  Sort bvsort, funsort, arrsort, boolsort;
 };
 
+class FreshTreeWalker : public TreeWalker {
```

Comment:
Above this line, please explain in detail what is the purpose of this walker (something like assigning a fresh Boolean variable to each occurrence of each term).